### PR TITLE
fix nans in bti dev head t when presen

### PR DESCRIPTION
**Pull request by dengemann**
This is to fix the issue reported by Joshua Bear that occurs when nans are found in the device to head transform. The patch enables reading the files that Joshua shared with us.

cc @agramfort

### CODE REVIEW

File: mne/io/bti/bti.py [hunk 1]
```diff
@@ -247,7 +247,13 @@ def _points_to_dig(points, n_idx_points, use_hpi):
 
 def _convert_coil_trans(coil_trans, dev_ctf_t, bti_dev_t):
     """Convert the coil trans."""
-    t = combine_transforms(invert_transform(dev_ctf_t), bti_dev_t,
+
+    dev_ctf_t_ = dev_ctf_t.copy()
+    has_nan = np.isnan(dev_ctf_t_['trans'])
+    if np.any(has_nan):
+        dev_ctf_t_['trans'] = np.identity(4)
```
Comment (agramfort) [inline mne/io/bti/bti.py]: I would have just done

```
if np.any(np.isnan(dev_ctf_t['trans'])):
     dev_ctf_t = dev_ctf_t.copy()
     dev_ctf_t['trans'] = np.identity(4)
```

**Comment by dengemann** [comment]
I have some archaic reflex not to overwrite variables in a local scope.
That often gave mysterious errors back then.

**Comment by agramfort** [comment]
At least avoid the systematic copy

**Comment by joshbear** [comment]
Hi Denis, 

Thank you for your quick work on this!

Your branch seems to have worked for reading in the file, though it looks like the problem might prevent writing the data as well. 

>>> raw.save('here-raw.fif')
Writing /research/data/here-raw.fif
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<string>", line 2, in save
  File "/Users/joshbear/bin/anaconda3/envs/mne/lib/python3.6/site-packages/mne-python/mne/utils.py", line 748, in verbose
    return function(*args, **kwargs)
  File "/Users/joshbear/bin/anaconda3/envs/mne/lib/python3.6/site-packages/mne-python/mne/io/base.py", line 1821, in save
    split_size, split_naming, part_idx, None, overwrite)
  File "/Users/joshbear/bin/anaconda3/envs/mne/lib/python3.6/site-packages/mne-python/mne/io/base.py", line 2272, in _write_raw
    reset_range, raw.annotations)
  File "/Users/joshbear/bin/anaconda3/envs/mne/lib/python3.6/site-packages/mne-python/mne/io/base.py", line 2446, in _start_writing_raw
    write_meas_info(fid, info, data_type=data_type, reset_range=reset_range)
  File "/Users/joshbear/bin/anaconda3/envs/mne/lib/python3.6/site-packages/mne-python/mne/io/meas_info.py", line 1412, in write_meas_info
    write_coord_trans(fid, info['dev_ctf_t'])
  File "/Users/joshbear/bin/anaconda3/envs/mne/lib/python3.6/site-packages/mne-python/mne/io/write.py", line 316, in write_coord_trans
    trans_inv = linalg.inv(trans['trans'])
  File "/Users/joshbear/bin/anaconda3/envs/mne/lib/python3.6/site-packages/scipy/linalg/basic.py", line 946, in inv
    a1 = _asarray_validated(a, check_finite=check_finite)
  File "/Users/joshbear/bin/anaconda3/envs/mne/lib/python3.6/site-packages/scipy/_lib/_util.py", line 238, in _asarray_validated
    a = toarray(a)
  File "/Users/joshbear/bin/anaconda3/envs/mne/lib/python3.6/site-packages/numpy/lib/function_base.py", line 1233, in asarray_chkfinite
    "array must not contain infs or NaNs")
ValueError: array must not contain infs or NaNs

**Comment by dengemann** [comment]
This should work now (also the writing).

File: mne/io/bti/bti.py [hunk 2]
```diff
@@ -245,6 +245,13 @@ def _points_to_dig(points, n_idx_points, use_hpi):
     return dig
 
 
+def _check_nan_dev_head_t(dev_ctf_t):
+    """Make sure we deal with nans."""
+    has_nan = np.isnan(dev_ctf_t['trans'])
+    if np.any(has_nan):
+        dev_ctf_t['trans'] = np.identity(4)
```
Comment (agramfort) [inline mne/io/bti/bti.py]: I would just add a logging message but besides LGTM

**Comment by joshbear** [comment]
Just had a chance to test the writing fix on two my empty room recordings and can confirm that it seems to be working. Thank you again for the quick fix! I'm hoping to be able to provide some contributions of my own instead of just pointing out the problem next time.

**Comment by agramfort** [comment]
thx @dengemann

**Pull request merged**